b2sdk/transfer/inbound/downloader/stats_collector.py

```diff
@@ -9,42 +9,71 @@
 ######################################################################
 
 import logging
-from dataclasses import dataclass, field
-from typing import List  # 3.7 doesn't understand `list` vs `List`
-from typing import Optional
+from dataclasses import (
+    dataclass,
+    field,
+)
+from time import perf_counter_ns
+from typing import (
+    Any,
+    List,
+    Optional,
+    Type,  # 3.7 doesn't understand `list` vs `List`
+)
 
 logger = logging.getLogger(__name__)
 
 
+class SingleStatsCollector:
+    def __init__(self):
+        self.storage: List[int] = []
+        self.started_perf_timer: Optional[int] = None
+
+    def __enter__(self) -> None:
+        self.started_perf_timer = perf_counter_ns()
+
+    def __exit__(self, exc_type: Type, exc_val: Exception, exc_tb: Any) -> None:
+        time_diff = perf_counter_ns() - self.started_perf_timer
+        self.storage.append(time_diff)
+        self.started_perf_timer = None
+
+    def __getitem__(self, item_or_slice):
+        return self.storage[item_or_slice]
+
+    @property
+    def has_any_entry(self) -> bool:
+        return len(self.storage) > 0
```

Comment:
```suggestion
        return self.storage
```

Reply:
Ok, we're going for the faster solution. Fair enough.

Reply:
Done.